modules/core/src/v2/coins/xlm.ts

```diff
@@ -541,7 +594,9 @@ export class Xlm extends BaseCoin {
     const addressDetails = this.getAddressDetails(address);
     const rootAddressDetails = this.getAddressDetails(rootAddress);
 
-    if (addressDetails.address !== rootAddressDetails.address) {
+    const baseAddress = this.isValidMuxedAddress(address) ? addressDetails.baseAddress : addressDetails.address;
```

Comment:
mentioned this in other PR, but why do we need to keep track of both `address` and `baseAddress`?
it seems like they should be mutually exclusive where only 1 address field will be used at any given time.

Reply:
it helps to know the base address when we're sending to a muxed one, because we need to be able to check the existence of the underlying account in order to decide whether to add a `createAccount` operation or a `payment` operation

Reply:
I simplified the checks by setting always `baseAddress` and `address` to AddressDetails. Now we can just check for `baseAddress` whenever we need it.